### PR TITLE
Better logging during shutdown and prevent shutdown race condition.

### PR DESCRIPTION
**Pull request by riteshnoronha**
@Ceraunograph @goose42 

This fixes the issue of the master process hanging around on shutdown. 

cc: @marcuswalser @alanbrent

### CODE REVIEW

**Comment by betamatt** [comment]
I'm not in a position to comment on the wisdom of forcing the process kill while it has threads trying to quiesce but an ensure couldn't hurt, unless you can guarantee that `worker_assignment_loop` can't throw anything else.

```
def worker_assignment_thread
  Thread.new do
    begin
      worker_assignment_loop
    rescue Chore::TerribleMistake => e
      Chore.logger.error 'PWS: Terrible mistake, shutting down Chore'
      Chore.logger.error e.message
      Chore.logger.error e.backtrace
    ensure
      Chore.logger.info "PWS: worker_assignment_thread ending"
      # WorkerAssignment thread is independent of the main thread.
      # The main thread is waiting on the consumer threads to join,
      # Due to some weird SQS behaviour, its possible that these threads
      # maynot join, and the assigment thread always exits, since it's 
      # nonblocking. This will ensure that the master process exits.
      Process.exit!(true)
    end    
  end
end
```

**Comment by marcuswalser** [comment]
Sup @betamatt!

File: lib/chore/strategies/worker/preforked_worker_strategy.rb [hunk 1]
```diff
@@ -23,6 +23,7 @@ def initialize(manager, opts = {})
         @self_read, @self_write = IO.pipe
         trap_signals(NUM_TO_SIGNAL, @self_write)
         @worker_manager = WorkerManager.new(create_master_socket)
+        at_exit { delete_socket_file}
```
Comment (riteshnoronha) [inline lib/chore/strategies/worker/preforked_worker_strategy.rb]: @Ceraunograph should prevent orphaned socket files

Comment (Ceraunograph) [inline lib/chore/strategies/worker/preforked_worker_strategy.rb]: It seem fine in theory, although I think it obscures shutdown behavior. Has this been tested to work?

**Comment by goose42** [comment]
LGTM. 👍

**Pull request merged**